examples_tests/06.MeshLoaders/main.cpp

```diff
@@ -7,81 +7,188 @@
+                queueFamilyIx_render = i;
+            if (queueFamilyProperties.queueFlags & nbl::video::IPhysicalDevice::EQF_COMPUTE_BIT)
+                queueFamilyIx_compute = i;
+            if (queueFamilyProperties.queueFlags & nbl::video::IPhysicalDevice::EQF_TRANSFER_BIT)
+                queueFamilyIx_transfer = i;
+
+            if (queueFamilyIx_render < queueFamilyPropertiesRange.size() && queueFamilyIx_compute < queueFamilyPropertiesRange.size() && queueFamilyIx_transfer < queueFamilyPropertiesRange.size())
+                break;
+            else if (i == queueFamilyPropertiesRange.size() - 1u)
+                assert(false);
+        }
+    }
+
+    nbl::core::set<uint32_t> families = { queueFamilyIx_render, queueFamilyIx_compute, queueFamilyIx_transfer };
+
+    nbl::video::ILogicalDevice::SCreationParams deviceParams;
+    deviceParams.queueParamsCount = families.size();
+    nbl::core::vector<nbl::video::ILogicalDevice::SQueueCreationParams> queuesParams;
+    queuesParams.reserve(families.size());
+
+    float priorities[32];
+    std::fill_n(priorities, 32, 1.f);
+
+    for (auto& family : families)
+    {
+        nbl::video::ILogicalDevice::SQueueCreationParams qp;
+        qp.count = NABLA_QUEUE_COUNT;
+        qp.familyIndex = family;
+        qp.flags = static_cast<nbl::video::IGPUQueue::E_CREATE_FLAGS>(0);
+        qp.priorities = priorities;
+        queuesParams.push_back(qp);
+    }
+    deviceParams.queueCreateInfos = queuesParams.data();
+
+    auto logicalDevice = gpuPhysicalDevice->createLogicalDevice(deviceParams);
+
+    if (!logicalDevice)
 		return 1; // could not create selected driver.
 
+    core::smart_refctd_ptr<nbl::video::IGPUCommandBuffer> commandBuffer[1];
+    auto commandPool = logicalDevice->createCommandPool(queueFamilyIx_render, nbl::video::IGPUCommandPool::ECF_RESET_COMMAND_BUFFER_BIT);
+    logicalDevice->createCommandBuffers(commandPool.get(), nbl::video::IGPUCommandBuffer::EL_PRIMARY, 1, commandBuffer);
 
-	//! disable mouse cursor, since camera will force it to the middle
-	//! and we don't want a jittery cursor in the middle distracting us
-	device->getCursorControl()->setVisible(false);
+    core::smart_refctd_ptr<nbl::asset::IAssetManager> assetManager;
+    {
+        nbl::core::smart_refctd_ptr<nbl::io::IFileSystem> fileSystem = nbl::core::make_smart_refctd_ptr<nbl::io::CFileSystem>("");
+        assetManager = core::make_smart_refctd_ptr<nbl::asset::IAssetManager>(std::move(fileSystem));
+    }
 
-	//! Since our cursor will be enslaved, there will be no way to close the window
-	//! So we listen for the "Q" key being pressed and exit the application
-	QToQuitEventReceiver receiver;
-	device->setEventReceiver(&receiver);
+    core::smart_refctd_ptr<nbl::scene::ISceneManager> sceneManager;
+    {
+        // how to create properly Scene Manager in new API?
+        // only interface class provided
+    }
```

Comment:
what about this one?

Reply:
we dont have scene manager yet

Reply:
ping @Erfan-Ahmadi for a rudimentary camera implementation